<h1>Activity 1 - PALMERS PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>
<h3>Analyst ; Jercel Mariscotes</h3>


In [1]:
#import libraries
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<h3>Converting Dataset(s) into DataFrame</h3>
<h3>Requirments</h3>
<ul>
    <li>
        Name of DataFrame
    </li>
    <li>
        File Path
    </li>
</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Checking for Null Values</h3>

In [6]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Filtering a Column</h3>

In [8]:
penguins['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [9]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [10]:
  penguins[penguins['species'] == 'Chinstrap']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [11]:
  chinstrap_df = penguins[penguins['species'] == 'Chinstrap']

In [12]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [13]:
 gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [14]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [15]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [16]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [17]:
(penguins['species'] =='Chinstrap') & (penguins['flipper_length_mm'] >= 18)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Length: 344, dtype: bool

In [18]:
penguins[(penguins['species'] =='Chinstrap') & (penguins['flipper_length_mm'] >= 18)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

<h2>FILLING UP NULL VALUES</h2>

In [19]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [20]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [21]:
penguins['bill_length_mm'].mean()
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [22]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [23]:
penguins['bill_depth_mm'].mean()
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [24]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [25]:
penguins['flipper_length_mm'].mean()
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [26]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [27]:
penguins['body_mass_g'].mean()
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [28]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [29]:
penguins['sex'].mode()
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [30]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>Changing the Data type</h2>

In [31]:
penguins['species'] = penguins['species'].astype('category')
penguins['year'] = penguins['species'].astype('category')

In [32]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    object  
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    object  
 7   year               344 non-null    category
dtypes: category(2), float64(4), object(2)
memory usage: 17.2+ KB


In [33]:
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [34]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h2>Explanatory Data Analytics</h2>
<hr>

<h3>Question #1</h3>
<h4>What sex have the longest bill length?</h4>

In [35]:
 penguins['bill_length_mm'].max()

np.float64(59.6)

In [36]:
penguins[penguins['bill_length_mm'] == 59.6]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
185  Gentoo  Biscoe            59.6           17.0              230.0   

     body_mass_g   sex    year  
185       6050.0  male  Gentoo

<h3>Insight:</h3>
<h4>Base on this , Gentoo Male have the longest bill length . As we can se here the longest bill length is 59.6 max.</h4>

<h2>Question #2</h2>
<h3>What was the highest body mass recorded?</h3>

In [37]:
 penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [38]:
penguins['kg'] = round (penguins['body_mass_g'] / 1000,1)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex       year   kg  
0    3750.000000    male     Adelie  3.8  
1    3800.000000  female     Adelie  3.8  
2    3250.000000  female     Adelie  3.2  
3    4201.754386    male     Adelie  4.2  
4    3450.000000  female     Adelie  3.4  
..           ...     ...        ...  ...  
339  4000.000000    male  Chinstrap  4.0  
340  3400.000000  female  Chinstrap  3.4  
341  3775.000000    male  Chinstrap  3.8  
342  4100.000000    male  Chinstrap  4.1  
343  3775.000000  female  Chinstrap  3.8  

[344 rows x 9 columns]

In [39]:
 penguins['kg'].max()

np.float64(6.3)

In [40]:
 penguins[penguins['kg'] == 6.3]


species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
169  Gentoo  Biscoe            49.2           15.2              221.0   

     body_mass_g   sex    year   kg  
169       6300.0  male  Gentoo  6.3

<h3>Insight:</h3>
<h4>The highest body mass redorded is again from Gentoo , with the 6300.0 body mass . In kg it is 6.3 kg max.</h4>

<h3>Question #3</h3>
<h4>How many female penguins are recorded in Torgersen Island? </h4>

In [41]:
penguins.mode()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Biscoe            41.1           17.0              190.0   

   body_mass_g   sex    year   kg  
0       3800.0  male  Adelie  3.6

In [42]:
penguins[penguins['kg'] == 6.3 ] ['island']      
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex       year   kg  
0    3750.000000    male     Adelie  3.8  
1    3800.000000  female     Adelie  3.8  
2    3250.000000  female     Adelie  3.2  
3    4201.754386    male     Adelie  4.2  
4    3450.000000  female     Adelie  3.4  
..           ...     ...        ...  ...  
339  4000.000000    male  Chinstrap  4.0  
340  3400.000000  female  Chinstrap  3.4  
341  3775.000000    male  Chinstrap  3.8  
342  4100.000000    male  Chinstrap  4.1  
343  3775.000000  female  Chinstrap  3.8  

[344 rows x 9 columns]

<h3>Insight:</h3>
<h4>As we can see here i used print function to filter the exact result. The result is 24 female recorded from Torgersen Island</h4>

<hr>

<h3>Question #4</h3>
<h4>What was the highest body mass recorded for the penguins in the dataset?</h4>

In [43]:
highest_body_mass_g = penguins['body_mass_g'].max()
highest_body_mass_g

np.float64(6300.0)

In [44]:
# Convert to kilograms for clarity
highest_body_mass_kg = round(highest_body_mass_g / 1000, 1)
highest_body_mass_kg

np.float64(6.3)

<h3>Insight:</h3>
<h4>As we can see here the highest body mass recorded was 6.3 kg</h4>

<h3>Question #5</h3>
<h4>How many male and female penguins are in the dataset?</h4>

In [45]:
gender_count = penguins['sex'].value_counts()
print(gender_count)

sex
male      179
female    165
Name: count, dtype: int64


<h3>Insight:</h3>
<h4>The population leans slightly towards males with a ratio of approximately 1.08 males for every female.</h4>

<h3>Question #6</h3>
<h4>Which species has the largest average flipper length?</h4>

In [46]:
largest_flipper = penguins.groupby('species')['flipper_length_mm'].mean()
print(largest_flipper)

species
Adelie       190.025758
Chinstrap    195.823529
Gentoo       217.055768
Name: flipper_length_mm, dtype: float64


<h3>Insight:</h3>
<h4>Gentoo penguins have the longest average flipper length, followed by Chinstrap and Adelie penguins. This suggests differences in swimming ability or ecological adaptation among the species.</h4>

<h3>Question #7</h3>
<h4>How many penguins were recorded in each year?</h4>

In [47]:
penguins['year'].value_counts().sort_index()

year
Adelie       152
Chinstrap     68
Gentoo       124
Name: count, dtype: int64

<h3>Insight:</h3>
<h4>As we can see here Adelie penguins are the most numerous, Chinstrap the least.</h4>

<h3>Question #8</h3>
<h4>Which species has the smallest average flipper length?</h4>

In [53]:
penguins.groupby('species')['flipper_length_mm'].mean()
penguins.groupby('species')['flipper_length_mm'].mean().idxmin()

'Adelie'

<h3>Insight:</h3>
<h4>The species with the shortest average flippers is Adelie, showing smaller swimming adaptations.</h4>

<h3>Question #9</h3>
<h4>How many entries are there in total for the Adelie species?</h4>

In [54]:
(penguins['species'] == 'Adelie').sum()

np.int64(152)

<h3>Insight:</h3>
<h4>There are 152 Adelie penguins, making them the most represented species.</h4>

<h3>Question #10</h3>
<h4>What is the minimum body mass recorded for the penguins?</h4>

In [56]:
penguins['body_mass_g'].min()

np.float64(2700.0)

<h3>Insight:</h3>
<h4>The lightest penguin recorded weighs 2700.0 g, showing the smallest body mass in the dataset.</h4>

<h3>Question #11</h3>
<h4>What is the average body mass for males versus females?</h4>

In [61]:
average_body_mass_by_sex = penguins.groupby('sex')['body_mass_g'].mean()
average_body_mass_by_sex

sex
female    3862.272727
male      4514.684407
Name: body_mass_g, dtype: float64

<h3>Insight:</h3>
<h4>On average, male individuals have a higher body mass (4514.68 g) than female individuals (3862.27 g), indicating a noticeable sex-based difference in body mass.</h4>

<h3>Question#12</h3>
<h4>Are there any missing values in the dataset?</h4>

In [63]:
missing_values = penguins.isnull().sum()
missing_values

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
kg                   0
dtype: int64

<h3>Insight:</h3>
<h4>There are no missing values in any of the dataset's columns, indicating that the data is complete and ready for analysis.</h4>

<h3>Question #13</h3>
<h4>What is the body mass in kilograms of the heaviest Adelie penguin?</h4>

In [65]:
penguins[penguins['species'] == 'Adelie']['body_mass_g'].max() / 1000

np.float64(4.775)

<h3>Insight:</h3>
<h4>The number 4.775 is stored as a high-precision decimal using NumPy's float64 format.</h4>

<h3>Question </h3>
<h4>What year is represented most commonly in the dataset?</h4>

In [66]:
penguins['year'].mode()[0]

'Adelie'

<h2>Data Aggregation</h2>

In [ ]:
species_island = penguins.groupby(['species','island'])['kg'].mean()
species_island

In [ ]:
 bill_lenght = penguins.groupby(['species','year']).count()
 bill_lenght

<h1>CORRELATION ANALYSIS</h1>
<hr>
<h3>Testing Correlation using Pearson R</h3>
<hr>
<p><b>Pearson R</b> can test correlation between two Variables ,given that both Variables are <b>NUMERICAL DATA</b></p>

In [ ]:
flipper_mass = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [ ]:
flipper_mass

<h2>Testing Correlation between Numerical and categorical Data.</h2>

In [ ]:
from scipy.stats import f_oneway

In [ ]:
#stats, p_value = penguins.f_oneway

#filter data
penguins[penguins['sex'] == 'male']['body_mass_g']

In [ ]:

 no_use, p_value = f_oneway(penguins[penguins['sex'] == 'male']['body_mass_g'],penguins[penguins['sex'] == 'female']['body_mass_g'])
p_value